vscode/microsoft-kiota/src/workspaceTreeProvider.ts

```diff
@@ -0,0 +1,59 @@
+import * as vscode from 'vscode';
+import * as path from 'path';
+import * as fs from 'fs';
+import { kiotaWorkspaceFile } from './constants';
+
+const workspaceJsonPath = path.join(vscode.workspace.workspaceFolders?.map(folder => folder.uri.fsPath).join('') || '', '.kiota', kiotaWorkspaceFile);
+
+export class WorkspaceTreeProvider implements vscode.TreeDataProvider<vscode.TreeItem> {
+
+    constructor(private context: vscode.ExtensionContext) {
+        void this.ensureKiotaDirectory();
+    }
+    async getChildren(element?: vscode.TreeItem): Promise<vscode.TreeItem[]> {
+        if (!element) {
+            return [new vscode.TreeItem(kiotaWorkspaceFile, vscode.TreeItemCollapsibleState.None)];
+        }
+        return [];
+    }
+
+    getTreeItem(element: vscode.TreeItem): vscode.TreeItem {
+        if (element) {
+            element.command = { command: 'kiota.workspace.openWorkspaceFile', title: "Open File", arguments: [vscode.Uri.file(workspaceJsonPath)], };
+            element.contextValue = 'file';
+        }
+        return element;
+    }
+
+    private async ensureKiotaDirectory(): Promise<unknown> {
+        if (!vscode.workspace.workspaceFolders || vscode.workspace.workspaceFolders.length === 0) {
+            await vscode.window.showErrorMessage(
+                vscode.l10n.t("No workspace folder found, open a folder first")
+            );
+            return;
+        }
+        const kiotaDir = path.dirname(workspaceJsonPath);
+        try {
+            await fs.promises.access(kiotaDir);
+        } catch (error) {
+            return error;
```

Comment:
Maybe just show this as a vscode error instead of returning it?

Reply:
Nice catch!